### PR TITLE
fix batching logic for processors

### LINKED ISSUE #186: Default processors calls `RepositoryInterface::add` for every object
Default processors calls `RepositoryInterface::add` for every object which is inefficient, as every `add` call triggers a flush on entity manager and breaks batching logic. Injecting a manager there and persisting them instead of adding to the repository will solve this issue.

### PR DESCRIPTION
**Pull request by oallain**
Fixes #186 

I put `batch_size` by default to 1, for flush after each item.

### CODE REVIEW

File: spec/Processor/ResourceProcessorSpec.php [hunk 1]
```diff
@@ -96,16 +100,18 @@ function it_can_process_an_array_of_country_data(
             ->shouldBeCalled();
 
         $this->process($dataset);
+        $entityManager->flush();
```
Comment (pamil) [inline spec/Processor/ResourceProcessorSpec.php]: I think there's `should*` missing here.

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->booleanNode('web_ui')->defaultTrue()->end()
-                        ->integerNode('batch_size')->defaultValue(0)->end()
+                        ->integerNode('batch_size')->defaultValue(1)->end()
```
Comment (pamil) [inline src/DependencyInjection/Configuration.php]: It's okay for now, but I'd think it over whether we want to have one as default, it's not the most performant setting.

Comment (oallain) [reply src/DependencyInjection/Configuration.php]: I put this value but I think 100 would be better. What do you think ?

**Comment by oallain** [comment]
Waiting for https://github.com/FriendsOfSylius/SyliusImportExportPlugin/pull/188

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (stefandoorn) [reply src/DependencyInjection/Configuration.php]: I think 100 is better than 1 indeed.

**Comment by stefandoorn** [comment]
Could you rebase on master (and remove both Twig commits)?

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (lsmith77) [reply src/DependencyInjection/Configuration.php]: agreed

**Comment by oallain** [comment]
Hello, i had :

- rebase on master
- set default batch_size to 100
- fix batching logic for JsonResourceImporter too
- remove Twig commit

Regards :smile:

**Comment by oallain** [comment]
Thank you for merged :smile:

**Pull request merged**